File: fix_navigation_final.py
#!/usr/bin/env python3
"""
Final script to fix Navigation.tsx syntax errors
"""
import re

def fix_navigation():
    with open('app/components/Navigation.tsx', 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Fix semicolon after onClick
    content = re.sub(r'onClick=\{closeAllMenus\};\>', 'onClick={closeAllMenus}\n                      >', content)
    
    # Fix service.icon syntax
    content = re.sub(r'<service \.icon', '<service.icon', content)
    
    # Fix closing parenthesis and semicolon
    content = re.sub(r'\)\)\};', '))}', content)
    
    # Fix any remaining semicolon issues
    content = re.sub(r';\s*>', '\n                      >', content)
    
    with open('app/components/Navigation.tsx', 'w', encoding='utf-8') as f:
        f.write(content)

if __name__ == "__main__":
    fix_navigation()

File: fix-navigation-final.js
#!/usr/bin/env node

import fs from 'fs';

const filePath = '/workspace/app/components/Navigation.tsx';
let content = fs.readFileSync(filePath, 'utf8');

// Fix semicolons in JSX attributes
content = content.replace(/onClick=\{([^}]+)\};/g, 'onClick={$1}');
content = content.replace(/className="([^"]*)"\s*;/g, 'className="$1"');
content = content.replace(/;\s*<\/div>/g, '</div>');
content = content.replace(/;\s*<\/button>/g, '</button>');
content = content.replace(/;\s*<\/Link>/g, '</Link>');
content = content.replace(/;\s*<\/nav>/g, '</nav>');
content = content.replace(/;\s*<\/ul>/g, '</ul>');
content = content.replace(/;\s*<\/li>/g, '</li>');

// Fix malformed JSX
content = content.replace(/;\s*<\/div>\s*\)\}/g, '</div>)}');
content = content.replace(/;\s*<\/div>\s*\)\}/g, '</div>)}');

// Fix missing closing tags
content = content.replace(/;\s*<\/div>\s*\)\}/g, '</div>)}');

// Clean up extra semicolons
content = content.replace(/;\s*;\s*/g, ';');
content = content.replace(/;\s*$/gm, '');

// Fix specific patterns
content = content.replace(/onClick=\{\(\) => toggleDropdown\('services'\)\};/g, "onClick={() => toggleDropdown('services')}");
content = content.replace(/onClick=\{\(\) => toggleDropdown\('it-services'\)\};/g, "onClick={() => toggleDropdown('it-services')}");
content = content.replace(/onClick=\{toggleMenu\};/g, "onClick={toggleMenu}");

fs.writeFileSync(filePath, content, 'utf8');
console.log('Fixed Navigation.tsx');